app/src/main/kotlin/com/exallium/todoapp/allnotes/AllNotesAdapter.kt

```diff
@@ -26,6 +51,7 @@ class AllNotesAdapter(val model: AllNotesModel,
     private var dataSubscription: Subscription? = null
     private var clickSubject = PublishSubject.create<Int>()
```

Comment:
something like this inside view holder:

```
fun clicks() = itemView.clicks().filter { position == -1 }.map { adapterPosition }
```

position is the position in the view holder.  when a view is recycled, it's set to -1 (by recycler view, not by us)

adapterPosition is the view's position in the adapter's data model.

So, we return a chain that filters out any clicks that are somehow propagated by things off screen, and emit the adapter position of the object clicked.

We can then subscribe to this in onCreateViewHolder:

```
fun onCreateViewHolder(...) {
   val viewHolder = ViewHolder.create(...)
   compositeSubscription.add(viewHolder.clicks().subscribe(onClickFn))
}
```